wget/wget.go

```diff
@@ -0,0 +1,202 @@
+package wget
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+	"path"
+	"strconv"
+	"sync"
+	"time"
+)
+
+// File represents infromation about file
+// we are going to download
+type File struct {
+	io.Reader       // To implement Reader interface
+	totalSize int64 // Total content size in bytes
+	counter   int64 // Total # of bytes transferred
+}
+
+type DownloadManager struct {
+	fileList        map[string]*File // key = link
+	fileListOrder   []string         // to order fileList map iteration
+	fileListLock    sync.Mutex       // to prevent race
+	printFinishedCh chan struct{}    // we use it to wait until last print fucntion will be terminated correctly
+}
+
+// NewDownloadManager is construct function
+func NewDownloadManager() *DownloadManager {
+	return &DownloadManager{
+		fileList:        make(map[string]*File),
+		fileListOrder:   make([]string, 0),
+		printFinishedCh: make(chan struct{}),
+	}
+}
+
+func ParseURL(link string) error {
+	_, err := url.ParseRequestURI(link)
+	if err != nil {
+		return err
+	}
+	resp, err := http.Get(link)
+	if err != nil {
+		return fmt.Errorf("wget: error connecting url :%v", err)
+		//fmt.Fprintf(os.Stderr, "wget: error connecting url :%v\n", err)
+	}
+	defer resp.Body.Close()
+
+	bodySize, err := strconv.ParseInt(resp.Header.Get("Content-Length"), 10, 0)
+	if err != nil {
+		return fmt.Errorf("wget: content length = %v :%v", bodySize, err)
+		//fmt.Fprintf(os.Stderr, "wget: error getting content length :%v\n", err)
```

Comment:
delete dead code always

Reply:
Have deleted dead code in commit 42bef00c07d5f5b947db74850f8533ee10ed0aa0